backend/de.metas.edi/src/main/java/de/metas/edi/api/impl/DesadvBL.java

```diff
@@ -1112,11 +1157,11 @@ private Optional<ITranslatableString> createSingleMsg(
 	}
 
 	@NonNull
-	private Optional<BigDecimal> computeDeliveredQtyInBPartnerUOM(
+	private Optional<Quantity> computeDeliveredQtyInBPartnerUOM(
```

Comment:
AFAIS this don't have to be optional, as it's never empty, or is  it?

Reply:
it is:

```Java
if (orderLine.getC_UOM_BPartner_ID() <= 0)
		{
			return Optional.empty();
		}
```